pyiceberg/io/pyarrow.py

```diff
@@ -1332,10 +1332,22 @@ def serialize(self, value: Any) -> bytes:
         return to_bytes(self.primitive_type, value)
 
     def update_min(self, val: Any) -> None:
-        self.current_min = val if self.current_min is None else min(val, self.current_min)
+        if self.current_min is None:
+            self.current_min = val
+        elif val is None:
+            # keep current_min
+            return
+        else:
+            self.current_min = min(val, self.current_min)
 
     def update_max(self, val: Any) -> None:
-        self.current_max = val if self.current_max is None else max(val, self.current_max)
+        if self.current_max is None:
+            self.current_max = val
+        elif val is None:
+            # keep current_max
+            return
+        else:
+            self.current_max = max(val, self.current_max)
```

Comment:
How about avoiding the `return`:
```suggestion
        if self.current_max is None:
            self.current_max = val
        elif val is not None:
            self.current_max = max(val, self.current_max)
```